File: Weld.MVC.Sample/Controllers/HomeController.cs
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Weld.Attributes;
using Weld.MVC.Sample.Models;
using Weld.Infra;

namespace Weld.MVC.Sample.Controllers
{
    public class HomeController : Controller
    {
        //
        // GET: /Home/

        public ActionResult Index()
        {
            return View();
        }

        
        [HttpGet]
        [AjaxMethod]
        public int Sum(int x,int y)
        {
            return x + y;
        }



        [HttpGet]
        [AjaxMethod]
        public void Store(int x)
        {
            Debug.WriteLine("Store hit with {0}", x);
        }

        [AjaxMethod]
        public JsonResult<Person> GetPerson()
        {
            var p = new Person {FirstName = "Jan", LastName = "De Vries"};

            var jsonResult = new JsonResult<Person>(Json(p));
            //jsonResult.JsonRequestBehavior = JsonRequestBehavior.AllowGet;
            return jsonResult;
        }
    }
}

File: Weld.MVC.Sample/Scripts/Weld/HomeController.ts
/// <reference path="../typings/jquery/jquery.d.ts" />
class HomeController
{
    Sum(x: number, y: number, callback: (data: number) => any)
    {
        var url = "/Home/Sum";
        var data = { x: x, y: y };
        $.ajax({
            url: url,
            data: data,
            success: callback,
        });
    }
    Store(x: number)
    {
        var url = "/Home/Store";
        var data = { x: x };
        $.ajax({
            url: url,
            data: data,
        });
    }
    GetPerson(callback: (data: Person) => any)
    {
        var url = "/Home/GetPerson";
        var data = {  };
        $.ajax({
            url: url,
            type : "POST",
            data: data,
            success: callback,
        });
    }
}

interface Person
{
    Id: number;
    FirstName: string;
    LastName: string;
}